src/components/featureHighlight/index.js

```diff
@@ -148,11 +148,19 @@ class FeatureHighlight extends BaseComponent {
     this.setTargetPosition();
   }
 
-  UNSAFE_componentWillReceiveProps(nextProps) {
-    this.setTargetPosition(nextProps);
+  static getDerivedStateFromProps(nextProps, prevState) {
+    const target = nextProps?.getTarget?.();
```

Comment:
Don't you want to check if 'getTarget' was changed before running this all method?

Reply:
I think this is practically what happens, node is null and return false, wdyt?

Reply:
Why node will return null for the target if it did not change?

Reply:
How do you check if `getTarget` has changed? we have access to `nextProps` and since it's a static method we don't have access to `this.props`

Reply:
You can put the target on state and compare prevState.target with nextPros.target